src/main/java/com/google/sps/servlets/GoServlet.java

```diff
@@ -61,33 +73,78 @@ public GoServlet() throws IOException {
    */
   public GoServlet(
       DirectionsClientFactory directionsClientFactory,
+      TasksClientFactory tasksClientFactory,
       String fakeApiKey,
       String fakeOrigin,
       String fakeDestination,
       List<String> fakeWaypoints) {
     this.directionsClientFactory = directionsClientFactory;
+    this.tasksClientFactory = tasksClientFactory;
     apiKey = fakeApiKey;
     origin = fakeOrigin;
     destination = fakeDestination;
     waypoints = fakeWaypoints;
   }
 
+  private List<Task> getTasks(TasksClient tasksClient) throws IOException {
+    List<TaskList> taskLists = tasksClient.listTaskLists();
+    List<Task> tasks = new ArrayList<>();
+    for (TaskList taskList : taskLists) {
+      tasks.addAll(tasksClient.listTasks(taskList));
+    }
+    return tasks;
+  }
+
+  /**
+   * Parses for locations in Tasks. Scope is public for testing purposes.
+   *
+   * @param prefix String which represents the prefix wrapped in square brackets to look for. (e.g.
+   *     Location if looking for [Location: ])
+   * @param tasks List of tasks to parse for locations from.
+   * @return List of strings representing the locations.
+   */
+  public List<String> getLocations(String prefix, List<Task> tasks) {
+    return tasks.stream()
+        .map(task -> task.getNotes())
+        .filter(Objects::nonNull)
+        .map(notes -> getLocation(prefix, notes))
+        .filter(place -> !place.equals("No " + prefix))
+        .collect(Collectors.toList());
+  }
+
+  private String getLocation(String prefix, String taskNotes) {
```

Comment:
While not required, some comments here would be nice to understand what the regex is filtering for

Reply:
Good call, regex can be confusing on its own.

Reply:
`RESOLVED`